### ISSUE TITLE
panic: sync: WaitGroup 1.0.0-beta3 

### ISSUE THREAD

**Issue by liv3d** [influxdata/influxdb#7149]
We managed to crash influxdb by running accidentally running the same drop command in two different console windows.

__System info:__ 
Centos 7 / VMWare
1 core ([E5-2620 v3](http://ark.intel.com/products/83352/Intel-Xeon-Processor-E5-2620-v3-15M-Cache-2_40-GHz))
8GB Memory

`/var/log/messages` output:

```
Aug 15 15:12:45 influxdb-1 influxd: [httpd] 127.0.0.1 - - [15/Aug/2016:15:12:45 +0000] "GET /ping HTTP/1.1" 204 0 "-" "InfluxDBShell/1.0.0-beta3" b85c38ac-62fa-11e6-b3fc-000000000000 75
Aug 15 15:12:45 influxdb-1 influxd: [query] 2016/08/15 15:12:45 DROP MEASUREMENT fluffy_capacity_manager_create_servers_success
Aug 15 15:12:47 influxdb-1 docker: t=2016-08-15T15:12:47+0000 lvl=info msg="Request Completed" logger=context userId=0 orgId=0 uname= method=GET path=/ status=302 remote_addr=10.10.97.220 time_ns=1ns size=29
Aug 15 15:12:47 influxdb-1 influxd: [tsm1] 2016/08/15 15:12:47 beginning full compaction of group 0, 1 TSM files
Aug 15 15:12:47 influxdb-1 influxd: [tsm1] 2016/08/15 15:12:47 compacting full group (0) /var/lib/influxdb/data/telegraf/default/55/000000032-000000061.tsm (#0)
Aug 15 15:12:48 influxdb-1 influxd: panic: sync: WaitGroup is reused before previous Wait has returned
Aug 15 15:12:48 influxdb-1 influxd: goroutine 3862801 [running]:
Aug 15 15:12:48 influxdb-1 influxd: panic(0xa4ac60, 0xc82186e0a0)
Aug 15 15:12:48 influxdb-1 influxd: /usr/local/go/src/runtime/panic.go:481 +0x3e6
Aug 15 15:12:48 influxdb-1 influxd: sync.(*WaitGroup).Wait(0xc820160d20)
Aug 15 15:12:48 influxdb-1 influxd: /usr/local/go/src/sync/waitgroup.go:129 +0x114
Aug 15 15:12:48 influxdb-1 influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).SetCompactionsEnabled(0xc820160d00, 0x0)
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:179 +0x2b7
Aug 15 15:12:48 influxdb-1 influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).DeleteSeriesRange(0xc820160d00, 0xc820268c10, 0x1, 0x1, 0x8000000000000000, 0x7fffffffffffffff, 0x0, 0x0)
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:602 +0x78
Aug 15 15:12:48 influxdb-1 influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).DeleteSeries(0xc820160d00, 0xc820268c10, 0x1, 0x1, 0x0, 0x0)
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:590 +0x69
Aug 15 15:12:48 influxdb-1 influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).DeleteMeasurement(0xc820160d00, 0xc82092f080, 0x2e, 0xc820268c10, 0x1, 0x1, 0x0, 0x0)
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:657 +0xbc
Aug 15 15:12:48 influxdb-1 influxd: github.com/influxdata/influxdb/tsdb.(*Shard).DeleteMeasurement(0xc820078dc0, 0xc82092f080, 0x2e, 0xc820268c10, 0x1, 0x1, 0x0, 0x0)
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/shard.go:414 +0xd2
Aug 15 15:12:48 influxdb-1 influxd: github.com/influxdata/influxdb/tsdb.(*Store).DeleteMeasurement.func2(0xc820078dc0, 0x0, 0x0)
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/store.go:484 +0x74
Aug 15 15:12:48 influxdb-1 influxd: github.com/influxdata/influxdb/tsdb.(*Store).walkShards.func1(0xc820d4b5c0, 0xc824db9ce0, 0xc820d4b680, 0xc820078dc0)
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/store.go:530 +0x8f
Aug 15 15:12:48 influxdb-1 influxd: created by github.com/influxdata/influxdb/tsdb.(*Store).walkShards
Aug 15 15:12:48 influxdb-1 influxd: /root/go/src/github.com/influxdata/influxdb/tsdb/store.go:536 +0x12a
Aug 15 15:12:48 influxdb-1 systemd: influxdb.service: main process exited, code=exited, status=2/INVALIDARGUMENT
Aug 15 15:12:48 influxdb-1 systemd: Unit influxdb.service entered failed state.
Aug 15 15:12:48 influxdb-1 systemd: influxdb.service failed.
Aug 15 15:12:48 influxdb-1 systemd: influxdb.service holdoff time over, scheduling restart.
Aug 15 15:12:48 influxdb-1 systemd: Started InfluxDB is an open-source, distributed, time series database.
Aug 15 15:12:48 influxdb-1 systemd: Starting InfluxDB is an open-source, distributed, time series database...
```

The query ran twice was:
```bash
influx --database telegraf --execute 'show measurements' |grep ^fluffy_ | xargs -n 1 -I '{}' influx --database telegraf --execute 'drop measurement {}'
```

**Comment by jsternberg** [comment]
I don't know how likely it is, but it might be possible to have this kind of error if compactions are disabled, enabled, and then disabled before the previous compactions are disabled.

**Comment by stevenh** [comment]
We didn't take any manual actions so unless this is being done automatically, I don't believe this is the cause.

**Comment by jsternberg** [comment]
It appears [this](https://github.com/influxdata/influxdb/blob/master/tsdb/engine/tsm1/engine.go#L626) is the code especially since that function name is mentioned in the stack trace. I was more just mentioning that to communicate with @jwilder so it would be easier for him to know where the problem was without having to look.

**Comment by e-dard** [comment]
I see what's going on here. I can have a stab at this.

**Comment by e-dard** [comment]
Fixed via #7165
